src/hotspot/share/gc/parallel/parMarkBitMap.cpp

```diff
@@ -173,27 +173,26 @@ ParMarkBitMap::iterate(ParMarkBitMapClosure* live_closure,
   // The bitmap routines require the right boundary to be word-aligned.
   const idx_t search_end = align_range_end(range_end);
 
-  idx_t cur_beg = find_obj_beg(range_beg, search_end);
-  while (cur_beg < range_end) {
-    const idx_t cur_end = find_obj_end(cur_beg, search_end);
-    if (cur_end >= range_end) {
-      // The obj ends outside the range.
-      live_closure->set_source(bit_to_addr(cur_beg));
```

Comment:
The caller of `iterate` expects `live_closure->source()` to be set at exit. It is in used.

This is not the case for the `iterate` taking two closures afaict, but nevertheless it seems part of the contract so it is surprising if some methods do and some do not.

The other changes seem to be okay.

Reply:
@albertnetymk mentioned that `do_addr` of `MarkAndUpdateClosure` properly updates the `source` of the closure.